src/nupic/regions/PyRegion.py

```
@@ -177,6 +177,15 @@ def getOutputElementCount(self, name):
     name: the name of the output
     """
 
+  def getAlgorithm(self):
+    """
+    Returns the underlying algorithm that is performing the computation.
+    This method should be overridden by the region subclass.
+
+    Returns None if there is no underlying algorithm for the region.
+    """
+    return None
```

I think it should throw an exception here rather than silently returning None.